src/middlewared/middlewared/plugins/filesystem.py

```diff
@@ -52,6 +62,24 @@ def resolve_cluster_path(self, path, ignore_ctdb=False):
         cluster_path = path.replace(FuseConfig.FUSE_PATH_SUBST.value, f'{FuseConfig.FUSE_PATH_BASE.value}/')
         return cluster_path
 
+    @accepts(Str('path'))
+    @returns()
+    def remove_directory(self, path):
```

Comment:
This will be abused and potentially cause problems. (`rmtree` is not removing just the directory). Since you added this for the API tests, I'd say remove this method and just use the `SSH_TEST(` method to remove the directory in the integration tests.